extensions/ql-vscode/src/query-history-serializater.ts

```diff
@@ -0,0 +1,178 @@
+import { pathExists, readFile, remove, mkdir, writeFile } from "fs-extra";
+import { dirname } from "path";
+
+import { showAndLogErrorMessage } from "./helpers";
+import {
+  asyncFilter,
+  getErrorMessage,
+  getErrorStack,
+} from "./pure/helpers-pure";
+import { CompletedQueryInfo, LocalQueryInfo } from "./query-results";
+import { QueryHistoryInfo } from "./query-history/query-history-info";
+import { QueryStatus } from "./query-status";
+import { QueryEvaluationInfo } from "./run-queries-shared";
+import { QueryResultType } from "./pure/legacy-messages";
+import { RemoteQueryHistoryItem } from "./data-serialization/source-schemas-types/remote-query-history-item";
+import { VariantAnalysisHistoryItem } from "./data-serialization/source-schemas-types/variant-analysis-history-item";
+import { App } from "./common/app";
+import { DisposableObject } from "./pure/disposable-object";
+
+type SerializableHistoryItems =
+  | LocalQueryInfo
+  | RemoteQueryHistoryItem
+  | VariantAnalysisHistoryItem;
+
+export class QueryHistorySerializer extends DisposableObject {
```

Comment:
I notice that this is a `DisposableObject`, but there is nothing for it to dispose. It's probably fine since in the future, we may want to add some `dispose` behaviour. And then I'm also wondering if `jsonValidator` needs any dispose behaviour either? It's maybe a little tricky since `jsonValidator` is owned by `App`, which is what should dispose it. Just trying to make sure nthing is missed.